kubernetes-tests/src/test/java/io/fabric8/kubernetes/client/mock/SubjectAccessReviewAuthTest.java

```diff
@@ -38,83 +42,112 @@
-      } catch (IOException e) {
-        logger.error("Invalid response received", e);
-      }
-
-      return reviewResponse;
-    }).once();
-
-    SubjectAccessReview reviewResponse = client.subjectAccessReviewAuth().create(review);
-
-    assertNotNull(reviewResponse);
-    assertEquals(true, reviewResponse.getStatus().getAllowed());
-
-  }
-
-  @Test
-  public void createLocalSubjectAccessReviewTest() {
-
-
-    KubernetesClient client = server.getClient();
-
-    LocalSubjectAccessReview review = new LocalSubjectAccessReviewBuilder().withNewSpec()
-      .withUser("admin-user")
-      .withNewResourceAttributes()
-      .withResource("pod")
-      .withVerb("create")
-      .endResourceAttributes()
-      .endSpec()
-      .build();
-
-    server.expect().post().withPath("/apis/authorization.k8s.io/v1/namespaces/test/localsubjectaccessreviews").andReply(200, recordedRequest -> {
-      ObjectMapper mapper = new ObjectMapper();
-      LocalSubjectAccessReview reviewResponse = null;
-      try {
-        reviewResponse = mapper.readValue(recordedRequest.getBody().readString(Charset.defaultCharset()), LocalSubjectAccessReview.class);
-        reviewResponse.setStatus(new SubjectAccessReviewStatus(true, false, "",""));
-      } catch (IOException e) {
-        logger.error("Invalid response received", e);
-      }
-
-      return reviewResponse;
-    }).once();
-
-    LocalSubjectAccessReview reviewResponse = client.subjectAccessReviewAuth()
-      .inNamespace("test").create(review);
-
-    assertNotNull(reviewResponse);
-    assertNotNull(reviewResponse.getMetadata());
-    assertEquals("test", reviewResponse.getMetadata().getNamespace());
-    assertEquals("test", reviewResponse.getSpec().getResourceAttributes().getNamespace());
-    assertEquals(true, reviewResponse.getStatus().getAllowed());
-
-  }
-}
+    private static final Logger logger = LoggerFactory.getLogger(ClusterRoleBindingCrudTest.class);
+    @Rule
+    public KubernetesServer server = new KubernetesServer();
+
+    @Test
+    public void createSubjectRulesReviewTest() {
+        KubernetesClient client = server.getClient();
+
+        createSubjectAccessReviewTest();
+
+        SelfSubjectRulesReview selfSubjectRulesReview = new SelfSubjectRulesReviewBuilder()
+                .withNewSpec()
+                .withNamespace("test")
+                .endSpec()
+                .withNewStatus()
+                .withResourceRules(new ResourceRuleBuilder()
+                        .withResourceNames("pod")
+                        .withVerbs("create", "list", "get", "delete").build()).endStatus().build();
+
+        SelfSubjectRulesReview rulesReviewResponse = client.subjectAccessReviewAuth().list().create(selfSubjectRulesReview);
+
+        assertNotNull(rulesReviewResponse);
+
+        ObjectMapper mapper = new ObjectMapper(new YAMLFactory());
+
+        SelfSubjectRulesReview reviewFromFile = null;
+
+        try {
```

Comment:
This is a unit test, you shouldn't be catching exceptions (add throws Exception to method signature)